lib/compilers/haskell.js

```diff
@@ -1,10 +1,53 @@
 var Compile = require('../base-compiler');
+var logger = require('../../lib/logger').logger;
+var child_process = require('child_process');
+var path = require('path');
 
 function compileHaskell(info, env) {
+    _ = require('underscore-node');
     var compiler = new Compile(info, env);
+
+    var usedStg = false;
     compiler.optionsForFilter = function (filters, outputFilename, userOptions) {
+        logger.warn("inside compiler.optionsForfilter");
+        usedStg = _.any(userOptions, function(opt) {
+            return opt.indexOf("-ddump-stg") > -1;
+        });
+
+        logger.warn("usedStg: ", usedStg);
+
+        if (usedStg) {
+            return ["-ddump-stg", "-ddump-to-file", "-o", this.filename(outputFilename)]
+            //return ["-ddump-stg", "-o", this.filename(outputFilename)]
+            // return ['-S', '-g', '-o', this.filename(outputFilename)];
+        }
         return ['-S', '-g', '-o', this.filename(outputFilename)];
     };
+
+    compiler.handlePostProcessResult = function (result, postResult, outputFilename) {
+        logger.warn("inside handlePostProcessResult");
+        logger.warn("usedStg: ", usedStg);
+        if (usedStg) {
+            const dirExploded = path.parse(outputFilename);
+            logger.warn("dirExploded: ");
+            logger.warn(dirExploded);
+            // HACK: this assumes that the input file name is `example`
+            const catStgFileCommand = 'cat ' + dirExploded.dir + '/' + 'example' + '.dump-stg';
+            logger.warn("catStgFileCommand: " + catStgFileCommand);
+
+            const stgData = child_process.execSync(catStgFileCommand).toString();
```

Comment:
How big can these files get? This should probably use a promise and exec.

Reply:
Ah, right. Well, it can get pretty big if the input code is large.